### DISCUSSION TITLE
How to set Caps Lock to ESC on tap and Extend on hold?

### DISCUSSION THREAD

**Discussion by Christensen-John** [DreymaR/BigBagKbdTrixPKL#79] [General]
I have been trying to figure out how to set caps lock to ESC when I tap it and then Extend on hold. In the settings window I have made the following changes:

* Layout to ANSI and Curl/Angle so that it should be set to Colemak-DH

I found [this discussion](https://github.com/DreymaR/BigBagKbdTrixPKL/discussions/64) and I tried the following:

`QWCLK = ESCAPE/Extend 		;`

I tried this in the following files:

* EPKL_Settings_Override
* EPKL_Layouts_Override
* Layouts/Colemak/Cmk-eD/Cmk_eD_ANS_CurlAngle

I am at a loss now and would love some help!

**Comment by DreymaR** [comment]
Unfortunately, you got the syntax slightly wrong. You'd have to set the key type as well, which you didn't. You're missing a `VK` or `VKey` on that line.

I recommend using the Layout/Settings dialog instead. Remove your previous changes first, so they don't override the correct changes you will be making (they probably wouldn't but just to be sure). Now, you can't select `ESC` in the "Map to..." dropdown boxes, but you can edit the line under `=` directly from `vcBSP` to `vcESC` and submit it.

Since this key isn't defined in layout.ini files, you can submit it for all layouts so it gets written to the `EPKL_Layouts_Override.ini` file. If you only want to affect your current layout for some reason, submit to layout instead.
